### ISSUE TITLE
Enable dependent input variable usage for compound launch configurations

### ISSUE THREAD

**Issue by MarcelKoller** [augustocdias/vscode-shell-command#95]
Is there any way to use dependent input variables in a compound launch config? I'll try to explain the issue with a simplified `launch.json`:
``` json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "gdb",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/build/bin/${input:selectApplication}",
            "cwd": "${workspaceFolder}/build/bin/",
            "MIMode": "gdb",
        },
        {
            "name": "python",
            "type": "debugpy",
            "request": "launch",
            "program": "${workspaceFolder}/build/python/${input:selectPython}",
            "args": [
                "${input:pythonArgs}",
            ],
        },
    ],
    "compounds": [
        {
            "name": "Python & gdb",
            "configurations": [
                "gdb",
                "python"
            ]
        }
    ],
    "inputs": [
        {
            "id": "selectApplication",
            "type": "command",
            "command": "shellCommand.execute",
            "args": {
                "command": "find . -type f -executable | grep -v test$ | sort -f",
                "cwd": "${workspaceFolder}/build/bin",
                "description": "Select the app to debug"
            },
        },
        {
            "id": "selectPython",
            "type": "command",
            "command": "shellCommand.execute",
            "args": {
                "command": "find . -type f -executable | sort -f",
                "cwd": "${workspaceFolder}/build/python",
                "description": "Select the python target to debug"
            },
        },
        {
            "id": "pythonArgs",
            "type": "command",
            "command": "shellCommand.execute",
            "args": {
                "command": "${input:selectPython} --someArgs",
                "cwd": "${workspaceFolder}/build/python",
                "description": "Select something"
            },
        },
    ]
}
```
When I launch `Python & gdb` I get promted (in this order) `selectPython`, `selectApplication` and finally `pythonArgs` which doesn't offer to select anything because `${input:selectPython}` seems to be resolved to "nothing". The standalone `python` launch target works. So it seems like the value of `${input:selectPython}` is "forgotten" because there was a switch to another launch configuration - at least that's my assumption. Any ideas how this could be fixed or worked around?

**Comment by MarcelRobitaille** [comment]
I think this should work. This is effectively "Dependent Input Variables Usage example:" from the readme.

**Comment by MarcelRobitaille** [comment]
Can you please make a reproducible example that does not require a folder of python binaries? Maybe just echo the input instead of executing it?

**Comment by MarcelRobitaille** [comment]
I have a suspicion. Could you please check if it's reproducible on my branch `test_95`?

**Comment by MarcelRobitaille** [comment]
If this still occurs in the next release, please re-open this issue.

**Comment by MarcelRobitaille** [comment]
Please try 1.12.0

**Comment by MarcelRobitaille** [comment]
That's very strange that it depends on timing. Is it possible that the command outputs nothing?

**Comment by MarcelRobitaille** [comment]
I don't see how there could possibly be a race condition here. The empty result is checked immediately after the command is executed, before any UI prompt stuff.

Can you please tell me the order in which the inputs are prompted, what you select each time, and how long you wait each time?

**Comment by MarcelRobitaille** [comment]
Does it show in the log the exact command that is run for `pythonArgs`?

**Comment by MarcelRobitaille** [comment]
@MarcelKoller Sorry, I was confused about something. Nevermind.

Do you know how to run VSCode extensions from source? If so, could you please try this branch: [task/95](https://github.com/augustocdias/vscode-shell-command/tree/task/95)?

**Comment by MarcelRobitaille** [comment]
> AFAIK it's not (easily) possible to control this using compound launch targets, right?

Could you do something using `preLaunchTask` and/or `dependsOn`?

> So it seems as of now I won't be able to use this configuration even if your proposed fix would solve the issue

Would you mind testing it anyway? I would like to fix this bug.

**Comment by MarcelRobitaille** [comment]
I suspect it's fixed in 1.13.0
